### PR TITLE
Add types to `_signals`, `test_signals`

### PR DESCRIPTION
**Pull request by TeamSpen210**
This is fairly straightforward, but I did extract a test helper method into a function. This allows `open_signal_receiver()` to be typed as producing an `AsyncIterator`, keeping `SignalReceiver` internal.

### CODE REVIEW

**Review by CoolCat467 — APPROVED** [review]
Looks good to me. If anything, the change removing `_pending_signal_count` could potentially cause issues if people are using this function even though it basically says not to because it was only there for the tests, but I think this is an acceptable change because:
1) People shouldn't be using said function anyways
2) Making it a separate function that isn't publicly exposed is a better solution for the tests than having it be a class function that technically could exposed

File: trio/_signals.py [hunk 1]
```diff
@@ -111,8 +118,17 @@ async def __anext__(self):
             return signum
 
 
+def get_pending_signal_count(rec: AsyncIterator[int]) -> int:
+    """Helper for tests, not public or otherwise used."""
+    # open_signal_receiver() always produces SignalReceiver, so cast.
```
Comment (A5rocks) [inline trio/_signals.py:123]: Out of date comment (you assert, not cast here).

**Review by A5rocks — APPROVED** [review]
Looks like a pretty simple file to type. Essentially no comments!

**Review by A5rocks — APPROVED** [review]
Looks good to me.

**Review by jakkdl — APPROVED** [review]

**Pull request merged**